Modules/_pickle.c

```diff
@@ -1311,6 +1339,74 @@ _Unpickler_ReadImpl(UnpicklerObject *self, char **s, Py_ssize_t n)
     return n;
 }
 
+/* Read `n` bytes from the unpickler's data source, storing the result in `buf`.
+ *
+ * This should only be used for non-small data reads where potentially
+ * avoiding a copy is beneficial.  This method does not try to prefetch
+ * more data into the input buffer.
+ *
+ * _Unpickler_Read() is recommended in most cases.
+ */
+static Py_ssize_t
+_Unpickler_ReadInto(UnpicklerObject *self, char *buf, Py_ssize_t n)
+{
+    assert(n != READ_WHOLE_LINE);
+
+    /* XXX add tests from ogrisel's use case */
```

Comment:
@ogrisel Do you remember what I might have meant here? :-) At some point I think you experimented with special file-like objects...

Reply:
As far as I remember, my main use case was a prototype for zero copy numpy serialization/deserialization on filesystem-backed file objects as implemented https://gist.github.com/ogrisel/a2b0e5ae4987a398caa7f9277cb3b90a

However I am not sure how it relates to readinto but it's probably necessary to be able to rebuild a large `PickleBuffer` from a pickle file without making an extra copy.